torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -324,17 +324,23 @@ def apply_fsdp(
         )
     fully_shard(model, **fsdp_config, reshard_after_forward=not pp_enabled)
 
-    if pp_enabled:
-        # TODO
-        # This PR https://github.com/pytorch/pytorch/pull/129519 added a safety check to avoid using 2D/3D DCP since
-        # without strided sharding, DCP can not safely support resharding for 2D/3D.  However, for PP to work, even
-        # without resharding, we load a seed-checkpoint and need to disable the safety mechanism.  This hack should be
-        # removed after strided sharding is landed in DCP.
-        for module in model.modules():
-            assert len(module._load_state_dict_pre_hooks) <= 1
+    # TODO: This PR https://github.com/pytorch/pytorch/pull/129519 added a safety check
```

Comment:
I think we should encapsulate the whole logic here to be a single function (i.e. lives under `parallelisms/utils.py` or `parallelisms/pipeline_utils.py`) and just call it here.